include/core/rtos/UARTTX.hpp

```diff
@@ -0,0 +1,191 @@
+#ifndef EVT_TX_UART_H
+#define EVT_TX_UART_H
+
+#include <HALf4/stm32f4xx.h>
+#include <core/io/UART.hpp>
+#include <core/io/pin.hpp>
+#include <core/rtos/Initializable.hpp>
+#include <core/rtos/Mutex.hpp>
+#include <core/rtos/Queue.hpp>
+#include <core/rtos/Semaphore.hpp>
+#include <core/rtos/Thread.hpp>
+#include <core/rtos/Threadx.hpp>
+#include <string>
+
+// All defines wrapped in ifndefs so we can set them externally.
+
+#ifndef UARTTX_QUEUE_MESSAGE_SIZE
+    #define UARTTX_QUEUE_MESSAGE_SIZE 16
+#endif // UARTTX_QUEUE_MESSAGE_SIZE
+
+#ifndef UARTTX_QUEUE_NUM_MESSAGES
+    #define UARTTX_QUEUE_NUM_MESSAGES 32
+#endif // UARTTX_QUEUE_NUM_MESSAGES
+
+#ifndef UARTTX_DEFAULT_STACK_SIZE
+    #define UARTTX_DEFAULT_STACK_SIZE 1024
```

Comment:
I mean if you want to have a bunch of threads that have the same default values, you could define them once and then you don't have to enter it in every time. It doesn't hurt to have these overwritable.

Reply:
I'm of the opinion that there should be only 1 way of doing anything. Having many ways to do the same thing makes the library confusing and difficult to use. @mjh9585 what do you think?

Reply:
I am not sure what you mean by multiple ways of doing things. I think overriding constants in this way is a nice way to add flexibility without having to think about it every time you use it resulting in more copied code no one knows why they are coping.

Reply:
I mean that you can change the value either by overriding the constant or by changing the value passed to the constructor. I think there should be only way to change the value.

Reply:
Good point. I reworked it so only the two that cannot be passed in as template parameters (THREADUART_QUEUE_MESSAGE_SIZE and THREADUART_QUEUE_NUM_MESSAGES) are able to be redefined, and I added a check to make sure that THREADUART_QUEUE_MESSAGE_SIZE is not defined above 16, because that's the max size.